packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -426,36 +430,41 @@ subtask(TASK_COMPILE_SOLIDITY_COMPILE_JOBS)
       const downloader = new CompilerDownloader(compilersCache);
       await downloader.downloadCompilersList();
 
+      const options = { concurrency: os.cpus().length, stopOnError: false };
       /**
        * This is a pretty hacky solution to avoid a major refactor and/or breaking changes.
        * The hack is the fact that we're just pre-downloading all of the necessary compilers
        * before actually running the job, so all the later code assumes the compiler is already
        * downloaded because if it wasn't then parallel compilation will break.
        */
-      await chunkedPromiseAll(
-        versionList.map((solcVersion) => {
-          return () =>
-            run(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD, {
-              solcVersion,
-              quiet: false,
-            });
-        }),
-        os.cpus().length
+      const downloadResults = await pMap(
+        versionList,
+        (solcVersion) => {
+          return run(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD, {
+            solcVersion,
+            quiet: false,
+          });
+        },
+        options
       );
 
-      const results = await chunkedPromiseAll<
-        CompilationSuccess | CompilationFailure
-      >(
-        compilationJobs.map((compilationJob, compilationJobIndex) => {
-          return () =>
-            run(TASK_COMPILE_SOLIDITY_COMPILE_JOB, {
-              compilationJob,
-              compilationJobs,
-              compilationJobIndex,
-              quiet,
-            });
-        }),
-        os.cpus().length
+      if (downloadResults instanceof Error) {
```

Comment:
I don't think this is necessary, since `p-map` will reject with the aggregate error (not return it).

Reply:
Yes, according to the docs it rejects with that error, hence the `await` throws

Reply:
Oh I see, I misread the docs